### PR TITLE
payara-3696-acc-executor-service

### PR DESCRIPTION
**Pull request by arjantijms**
Payara-3696 Don't init executor service when in ACC

Hint for reviewers; review per individual commit. The first commit is adding a [debug option](https://github.com/payara/Payara/pull/3936/commits/393e1934f041004faba4fd5e2942c0987981f5b2#diff-d1d42039c38aeeee99906d790f64d73aR904) to the ACC start, since it's otherwise quite difficult to debug the agent code for the ACC.

The second commit fixes the exception during startup of the ACC by not initialising the executor service on the client-side.

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
> I'm afraid this may lead to null pointer exceptions down the road at other places.

I think we'd seen those already then, as initialisation was already skipped, just with an NPE before.

**Comment by arjantijms** [comment]
Jenkins test please

File: nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java [hunk 1]
```diff
@@ -141,37 +136,64 @@ public void event(Event event) {
...
+            // The application client container doesn't use the executor service locally for now.
+            // But create a "mock" executor to be absolutely sure.
             threadPoolExecutor = new ThreadPoolExecutor(
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorCorePoolSize()),
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorMaxPoolSize()),
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTime()),
-                    TimeUnit.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTimeUnit()),
-                    new LinkedBlockingQueue<>(threadPoolExecutorQueueSize),
-                    (Runnable r) -> new Thread(r, "payara-executor-service-task"));
+                0,
```
Comment (MattGill98) [inline nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java]: Is it possible to move the configured properties to local variables and override them if isAcc passes?

**Pull request merged**